ambry-server/src/main/java/com.github.ambry.server/AmbryRequests.java

```diff
@@ -459,6 +469,84 @@ public void handleDeleteRequest(Request request) throws IOException, Interrupted
             metrics.deleteBlobTotalTimeInMs, null, null, totalTimeSpent));
   }
 
+  @Override
+  public void handleTtlUpdateRequest(Request request) throws IOException, InterruptedException {
+    TtlUpdateRequest updateRequest =
+        TtlUpdateRequest.readFrom(new DataInputStream(request.getInputStream()), clusterMap);
+    long requestQueueTime = SystemTime.getInstance().milliseconds() - request.getStartTimeInMs();
+    long totalTimeSpent = requestQueueTime;
+    metrics.updateBlobTtlRequestQueueTimeInMs.update(requestQueueTime);
+    metrics.updateBlobTtlRequestRate.mark();
+    long startTime = SystemTime.getInstance().milliseconds();
+    TtlUpdateResponse response = null;
+    try {
+      ServerErrorCode error =
+          validateRequest(updateRequest.getBlobId().getPartition(), RequestOrResponseType.TtlUpdateRequest, false);
+      if (error != ServerErrorCode.No_Error) {
+        logger.error("Validating TtlUpdateRequest failed with error {} for request {}", error, updateRequest);
+        response = new TtlUpdateResponse(updateRequest.getCorrelationId(), updateRequest.getClientId(), error);
+      } else {
+        BlobId convertedStoreKey =
+            (BlobId) getConvertedStoreKeys(Collections.singletonList(updateRequest.getBlobId())).get(0);
+        MessageFormatInputStream stream =
+            new TtlUpdateMessageFormatInputStream(convertedStoreKey, convertedStoreKey.getAccountId(),
+                convertedStoreKey.getContainerId(), updateRequest.getExpiresAtMs(), updateRequest.getOperationTimeInMs());
+        MessageInfo info =
+            new MessageInfo(convertedStoreKey, stream.getSize(), false, true, updateRequest.getExpiresAtMs(),
+                convertedStoreKey.getAccountId(), convertedStoreKey.getContainerId(), updateRequest.getOperationTimeInMs());
+        MessageFormatWriteSet writeset = new MessageFormatWriteSet(stream, Collections.singletonList(info), false);
+        Store store = storageManager.getStore(updateRequest.getBlobId().getPartition());
+        store.updateTtl(writeset);
+        response = new TtlUpdateResponse(updateRequest.getCorrelationId(), updateRequest.getClientId(),
+            ServerErrorCode.No_Error);
+        if (notification != null) {
+          notification.onBlobReplicaUpdated(currentNode.getHostname(), currentNode.getPort(), convertedStoreKey.getID(),
+              BlobReplicaSourceType.PRIMARY, UpdateType.TTL_UPDATE, info);
+        }
+      }
+    } catch (StoreException e) {
+      boolean logInErrorLevel = false;
+      if (e.getErrorCode() == StoreErrorCodes.ID_Not_Found) {
```

Comment:
This is a lot of if/then/else statements.  If feasible, suggest adding method like:
Metrics.incrementErrorCount(StoreErrorCodes)
and calling that here.

Reply:
Each of these handler functions have a different definition of when to log at the error level (which is also something this if/else does). Any ideas on how to combine everything?

Reply:
I see.  Then never mind.